src/main/java/com/myDomain/myLibrary/RandomCircles.java

```diff
@@ -0,0 +1,58 @@
+package com.myDomain.myLibrary;
+
+import processing.core.*;
+import org.apache.commons.math3.random.*;
+
+/**
+ * This is an example class within library myLibrary.
+ * Make sure you rename this class as well as the name of the example package 'com.myDomain.myLibrary'
+ * to your own library naming convention.
+ */
+
+public class RandomCircles {
+	
+	// parent is a reference to the parent sketch
+	PApplet parent;
+
+	/**
+	 * a Constructor, usually called in the setup() method in your sketch to
+	 * initialize and start the Library.
+	 * 
+	 * @param theParent the parent PApplet
+	 */
+	public RandomCircles(PApplet theParent) {
+		this.parent = theParent;
+	}
+
+
+	/**
+	 * drawCircles draws circles
+	 * where their location is according to the Sobol Sequence,
+	 * a statistical method found in Apache Commons Math.
+	 *
+	 * @param numCircles the number of circles to draw
+	 */
+	public void drawCircles(int numCircles) {
```

Comment:
maybe mention in the comments that standard processing functions now need to be preceded with `this.parent.`

Reply:
Good idea. I added a comment next to the code. is what I added clear?

Reply:
Considering the level that the other comments are at, I would add the explanation that these functions are not available because we are not inside the PApplet class

Reply:
Sure. Can you check again the file? thank you.